test/common/schemaform/definitions/address.unit.spec.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { findDOMNode } from 'react-dom';
+import { expect } from 'chai';
+import ReactTestUtils from 'react-addons-test-utils';
+
+import { DefinitionTester } from '../../../util/schemaform-utils.jsx';
+import { schema, uiSchema } from '../../../../src/js/common/schemaform/definitions/address';
+
+describe('Schemaform definition address', () => {
+  it('should render address', () => {
+    const s = schema(false);
+    const uis = uiSchema();
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={s}
+          uiSchema={uis}/>
+    );
+
+    const formDOM = findDOMNode(form);
+
+    // Count the form elements
+    const inputs = formDOM.querySelectorAll('input');
+    const selects = formDOM.querySelectorAll('select');
+    expect(inputs.length).to.equal(4);
+    expect(selects.length).to.equal(2);
+
+    // Postal code should be small
+    expect(inputs[inputs.length - 1].classList.contains('usa-input-medium')).to.be.true;
+  });
+
+  it('should have required inputs if required', () => {
+    const s = schema(true);
+    const uis = uiSchema();
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={s}
+          uiSchema={uis}/>
+    );
+
+    const formDOM = findDOMNode(form);
+
+    // Ideally, we'd get the required inputs, not the <span>s denoting required
+    //  fields but this doesn't work.
+    // const requiredInputs = formDOM.querySelectorAll('input[required=true]');
+    const requiredInputs = formDOM.querySelectorAll('label span.schemaform-required-span');
+    expect(requiredInputs.length).to.not.equal(0);
+  });
+
+  it('should update labels and state selection conditionally', () => {
+    const s = schema(false);
+    const uis = uiSchema();
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={s}
+          uiSchema={uis}/>
+    );
+
+    const formDOM = findDOMNode(form);
+    const find = formDOM.querySelector.bind(formDOM);
+
+    // By default, the Country is USA, so the postalCode label should be 'ZIP Code'
+    //  and the State label should be 'State' and the field should be a dropdown
+    const labels = ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'label');
+    const postalCodeLabel = labels.find(label => label.htmlFor === 'root_postalCode');
+    const stateLabel = labels.find(label => label.htmlFor === 'root_state');
+    const stateField = formDOM.querySelectorAll('#root_state');
+
+    expect(postalCodeLabel.textContent === 'ZIP Code');
+    expect(stateLabel.textContent === 'State');
+    expect(stateField.tagName === 'select');
```

Comment:
It would also be good to test that the state select box options are constrained correctly, depending on country. You could probably just check for particular states for CAN, USA, and MEX.

Also should test that setting a military city (APO/FPO) results in the military state codes being shown in the select box.